jormungandr/src/network/subscription.rs

```diff
@@ -105,12 +107,40 @@ where
     tokio::spawn(
         inbound
             .for_each(move |gossip| {
-                debug!(logger, "received gossip: {:?}", gossip);
-                state.topology.update(gossip.into_nodes());
+                trace!(logger, "received gossip: {:?}", gossip);
+                let nodes = gossip
+                    .into_nodes()
+                    .filter(|node| filter_gossip_node(node, &state.config, &logger));
+                state.topology.update(nodes);
                 Ok(())
             })
             .map_err(move |err| {
                 info!(err_logger, "gossip subscription stream failure: {:?}", err);
             }),
     )
 }
+
+fn filter_gossip_node(node: &Node, config: &Configuration, logger: &Logger) -> bool {
+    let accepted = if config.allow_private_addresses {
+        node.has_valid_address()
+    } else {
+        node.is_global()
+    };
+    if !accepted {
+        if let Some(addr) = node.address() {
+            warn!(
+                logger,
+                "node {} has unacceptable IP address {}, pruned from gossip",
+                node.id(),
+                addr.ip()
+            );
```

Comment:
We will get a lot of logs, could we filter them all and then says something like:

filtering out these `[addresses]` from the gossips from `address` of the one who sent us these gossips?

Reply:
Done in a follow-up comment. The peer address should be already added to the `Logger` context of the connection. If not, it should be fixed elsewhere.